compat/winansi.c

```diff
@@ -573,6 +573,8 @@ static void detect_msys_tty(int fd)
 	if (!NT_SUCCESS(NtQueryObject(h, ObjectNameInformation,
 			buffer, sizeof(buffer) - 2, &result)))
 		return;
+	if (result <= sizeof(*nameinfo))
```

Comment:
While this works currently (on my Windows version, `result` is larger than `sizeof(*nameinfo)`, I am worried that this is fragile: nothing prevents a future Windows version from setting `result` to exactly the size of the `struct`. I would much prefer this:

```c
if (result < sizeof(*nameinfo) || !nameinfo->name.Buffer)
    return;
```

That should be more robust. Would you agree?

Reply:
I think I might extend that even further with
```suggestion
	if (result <= sizeof(*nameinfo) || !nameinfo->name.Buffer || !nameinfo->Name.Length)
```
just in case `Buffer` points to somewhere, but is an empty string too

Reply:
Change has been implemented, please re-review, Thanks

Reply:
Realized that this would make it go over 80 characters, re-pushed an indented version